pkg/pushsync/pushsync.go

```diff
@@ -241,20 +241,17 @@ func (ps *PushSync) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream)
 	}
 
 	// forwarding replication
-	storedChunk := false
+	storerNode := false
 	defer func() {
-		if !storedChunk {
-			if ps.warmedUp() && ps.topologyDriver.IsWithinDepth(chunkAddress) {
-				verifiedChunk, err := ps.validStamp(chunk, ch.Stamp)
+		if !storerNode && ps.warmedUp() && ps.topologyDriver.IsWithinDepth(chunkAddress) {
+			verifiedChunk, err := ps.validStamp(chunk, ch.Stamp)
+			if err != nil {
+				ps.metrics.InvalidStampErrors.Inc()
+				ps.logger.Warningf("pushsync: forwarder, invalid stamp for chunk %s", chunkAddress.String())
+			} else {
```

Comment:
since we are in a closure, you can just `return` inside the error block, and remove the `else` clause to reduce nesting